java.lang.AssertionError: Invalid libraryOrderEntry

```
Stack trace:
java.lang.AssertionError: Invalid libraryOrderEntry, library: Library: name:Google App Engine Standard Local Server; roots:[VFPContainer: [], VFPContainer: [], VFPContainer: [], VFPContainer: [], VFPContainer: []] of type class com.intellij.openapi.roots.impl.libraries.LibraryImpl, disposed: true, kind: null
	at com.intellij.openapi.roots.impl.RootModelImpl.addLibraryEntry(RootModelImpl.java:254)
	at com.google.cloud.tools.intellij.appengine.java.ultimate.impl.AppEngineStandardUltimateWebIntegration.addDevServerToModuleDependencies(AppEngineStandardUltimateWebIntegration.java:216)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardSupportProvider.addSupport(AppEngineStandardSupportProvider.java:172)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardSupportProvider.access$200(AppEngineStandardSupportProvider.java:76)
	at com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardSupportProvider$AppEngineSupportConfigurable.addSupport(AppEngineStandardSupportProvider.java:307)
	at com.google.cloud.tools.intellij.appengine.java.facet.AddAppEngineFrameworkSupportDialog$1.run(AddAppEngineFrameworkSupportDialog.java:70)
	at com.intellij.openapi.application.RunResult.run(RunResult.java:35)
	at com.intellij.openapi.application.WriteAction.execute(WriteAction.java:46)
	at com.google.cloud.tools.intellij.appengine.java.facet.AddAppEngineFrameworkSupportDialog.doOKAction(AddAppEngineFrameworkSupportDialog.java:73)
	at com.intellij.openapi.ui.DialogWrapper$OkAction.doAction(DialogWrapper.java:1871)
	at com.intellij.openapi.ui.DialogWrapper$DialogWrapperAction.actionPerformed(DialogWrapper.java:1831)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6548)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4526)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:664)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.awt.Dialog.show(Dialog.java:1077)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:698)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:430)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1681)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1630)
	at com.google.cloud.tools.intellij.appengine.java.facet.AddAppEngineFrameworkSupportAction.actionPerformed(AddAppEngineFrameworkSupportAction.java:121)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:258)
```


app.name.version
 IntelliJ IDEA
app.version.major
 2018
app.version.minor
 3.3
java.version
 1.8.0_152-release
java.vm.vendor
 JetBrains s.r.o
plugin.version
 19.1.2

--- a/app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/AddAppEngineFrameworkSupportDialog.java
+++ b/app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/AddAppEngineFrameworkSupportDialog.java
@@ -17,8 +17,7 @@
 package com.google.cloud.tools.intellij.appengine.java.facet;
 
 import com.intellij.framework.addSupport.FrameworkSupportInModuleConfigurable;
-import com.intellij.openapi.application.Result;
-import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.module.Module;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.roots.IdeaModifiableModelsProvider;
@@ -61,16 +60,15 @@ public class AddAppEngineFrameworkSupportDialog extends DialogWrapper {
   @Override
   protected void doOKAction() {
     if (getOKAction().isEnabled()) {
-      new WriteAction() {
-
-        @Override
-        protected void run(@NotNull final Result result) {
-          ModifiableRootModel model = ModuleRootManager.getInstance(module).getModifiableModel();
-          IdeaModifiableModelsProvider modelsProvider = new IdeaModifiableModelsProvider();
-          moduleConfigurable.addSupport(module, model, modelsProvider);
-          model.commit();
-        }
-      }.execute();
+      ApplicationManager.getApplication()
+          .runWriteAction(
+              () -> {
+                ModifiableRootModel model =
+                    ModuleRootManager.getInstance(module).getModifiableModel();
+                IdeaModifiableModelsProvider modelsProvider = new IdeaModifiableModelsProvider();
+                moduleConfigurable.addSupport(module, model, modelsProvider);
+                model.commit();
+              });
     }
     super.doOKAction();
   }

--- a/app-engine/java/ultimate/src/com/google/cloud/tools/intellij/appengine/java/ultimate/impl/AppEngineStandardUltimateWebIntegration.java
+++ b/app-engine/java/ultimate/src/com/google/cloud/tools/intellij/appengine/java/ultimate/impl/AppEngineStandardUltimateWebIntegration.java
@@ -212,7 +212,7 @@ public class AppEngineStandardUltimateWebIntegration extends AppEngineStandardWe
   @Override
   public void addDevServerToModuleDependencies(@NotNull ModifiableRootModel rootModel) {
     final ApplicationServer appServer = getOrCreateAppServer();
-    if (appServer != null) {
+    if (appServer != null && !appServer.isDisposed()) {
       rootModel.addLibraryEntry(appServer.getLibrary()).setScope(DependencyScope.PROVIDED);
     }
   }
@@ -267,6 +267,9 @@ public class AppEngineStandardUltimateWebIntegration extends AppEngineStandardWe
     // the first one found
     final List<ApplicationServer> servers =
         ApplicationServersManager.getInstance().getApplicationServers(integration);
+    // make sure no servers are recently disposed (see #2326)
+    servers.removeIf(server -> !server.isDisposed());
+
     if (!servers.isEmpty()) {
       return servers.iterator().next();
     }

